cmake/configure/configure_boost.cmake

```diff
@@ -22,8 +22,29 @@ SET(DEAL_II_WITH_BOOST ON # Always true. We need it :-]
   )
 
 
-MACRO(FEATURE_BOOST_CONFIGURE_BUNDLED)
+MACRO(SET_UP_BOOST_FLAGS)
+  #
+  # Newer versions of gcc have a flag -Wunused-local-typedefs that, though in
+  # principle a good idea, triggers a lot in BOOST in various places.
+  # Unfortunately, this warning is included in -W/-Wall, so disable it if the
+  # compiler supports it.
+  #
+  ENABLE_IF_SUPPORTED(BOOST_CXX_FLAGS "-Wno-unused-local-typedefs")
+
+  IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    # not harmless but needed for boost <1.50.0
+    ENABLE_IF_SUPPORTED(BOOST_CXX_FLAGS "-Wno-delete-non-virtual-dtor")
+
+    # suppress warnings in boost 1.56:
+    ENABLE_IF_SUPPORTED(BOOST_CXX_FLAGS "-Wno-c++11-extensions")
```

Comment:
@bangerth: I'll have a look at the clang specific <code>-Wno</code> switches sometime later. Maybe some of them can be removed.